### PR TITLE
Adds taunting as a replacement for spin/flip in style meter and parrying

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Replaces spin/flip's uses in mining's style meter and bitrunner's projectile parrying and instead uses a new emote, taunt. It also does not play a sound effect for emoting, only when you successfully block a projectile.

The parrying time from the flip was 1.4 seconds, with taunting it is now 0.9 seconds. Taunting also has a cooldown of 1.5 seconds between uses.

https://www.youtube.com/watch?v=cJGuEqNhqUs

https://github.com/user-attachments/assets/1c6bf8b2-6a0d-4ae2-9b5b-9c6e619e76d6

## Why It's Good For The Game

Spin and flip are emotes that get a little boring and repetitive, which makes its complete overuse quite annoying to see.
Compared to spin, taunting is a quick turn, which stacked with the cooldown, makes it a shorter and un-overusable emote.

It also has a cooldown of 1.5s between hits, so players now actually have a limit to how much they can parry/style. Currently if you have some way to regenerate stamina damage, you're pretty set to spam parry all projectiles at essentially no cost, since emotes cost nothing to use, removing the limit of having to actually time it.
I wouldnt say falling over is necessarily a downside since anyone who uses these items for a while can quickly figure out exactly hwo to maximize parrying time.

## Changelog

:cl:
add: Added Taunting, a faster and cooldowned version of the Spin emote.
balance: Wizards blocking projectiles with Transparence and the bitrunner matrix skillchip now have a visible effect of deflecting the projectile.
balance: The bitrunner skillchip now uses taunt instead of flip.
balance: The style meter now uses taunting instead of flips and spins.
/:cl:

### CODE REVIEW

**Comment by Paxilmaniac** [comment]
it needs the pizza tower taunt sound effect when you block a projectile

**Comment by JohnFulpWillard** [comment]
I wouldve but it's copyrighted

**Comment by Eronymun** [comment]
Oh this doesn't effect the lavaland only style meter parry does it?

**Comment by JohnFulpWillard** [comment]
Yes it does

**Comment by Eronymun** [comment]
Flipping isn't actually used for parries with the style meter, just timed right clicks. For the the wizard and bitrunner dodges this makes sense since it's a balance that moves one emote to another but this would be an unfairly heavy handed nerf to a already pretty rarely used lavaland exclusive feature.

**Comment by Eronymun** [comment]
> It affects the points but it doesnt parry itself, it still uses the system it's got since it doesn't use emotes.

Ah okay I see that makes sense. I still think it's a bit of referential flavor lost but that's fair.

**Comment by Ghommie** [comment]
I'm against the nerf to the flipping skillchip with all my heart. It isn't even powerful to warrant a nerf, and most sources of stamina regen don't keep up with its cost anyway, while also having occasional side-effects. Furthermore, this PR will nerf it to be even worse than it originally was at the time it was added (0.8 seconds immunity with a cooldown of 0.8 seconds).

File: code/modules/library/skill_learning/generic_skillchips/matrix_flip.dm
Comment (Ghommie) [inline code/modules/library/skill_learning/generic_skillchips/matrix_flip.dm:1]: Rename the file too.

**Comment by Ghommie** [comment]
CONFLICTS.

**Pull request merged**